### ISSUE TITLE
`explicit-override` false positive when overriding property with setter

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#15900]
```py
# mypy: enable-error-code=explicit-override

from typing_extensions import override


class Foo:
    @property
    def foo(self) -> int:
        return 1

    @foo.setter
    def foo(self, value: int) -> None:
        ...


class Bar(Foo):
    @property  # error: explicit-override
    @override
    def foo(self) -> int:
        return 2

    @foo.setter
    def foo(self, value: int) -> None:
        ...
```
[playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=strict%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-subclassing-any%2Cwarn-return-any%2Cwarn-unreachable&gist=da8ab80a737c9e1fee80e0bdd802993d)

putting `@override` on the setter does not fix it (and it shouldn't be required anyway, as that would be inconsistent with other decorators such as `@abstractmethod`)
